aie_runtime_lib/CMakeLists.txt

```diff
@@ -52,7 +52,29 @@ function(add_aie_runtime_libs arch)
 
 endfunction()
 
+add_custom_target(aie-ukernels-libs ALL)
+
+function(add_aie_ukernels_libs arch)
+  if(DEFINED VITIS_ROOT)
+    if(EXISTS "${PROJECT_SOURCE_DIR}/tools/chess-clang/xchesscc_wrapper")
+        add_custom_target(${arch}_ukernels ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/mm.o)
+        add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/mm.o
+                      COMMAND ${PROJECT_SOURCE_DIR}/tools/chess-clang/xchesscc_wrapper ${arch} -I ${VITIS_ROOT}/aietools/include -c ${CMAKE_CURRENT_SOURCE_DIR}/mm.cc -o ${CMAKE_CURRENT_BINARY_DIR}/mm.o
+                      DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/mm.cc)
+        add_dependencies(aie-ukernels-libs ${arch}_ukernels)
+    
+        install(FILES ${CMAKE_CURRENT_BINARY_DIR}/mm.o DESTINATION ${CMAKE_INSTALL_PREFIX}/aie_ukernel_lib/)
+    else()
+      message(STATUS "Could not find xchesscc_wrapper inside mlir-aie's build directory")
+    endif()
+  else()
+      message(STATUS "Tool 'xchesscc' not found. Skipping compilation of microkernel .o files.")
+  endif()
+
+endfunction()
+
 add_subdirectory(AIE)
 add_subdirectory(AIE2)
+add_subdirectory(Ukernels)
```

Comment:
I am not sure if aie kernels belong in a subfolder of the aie_runtime_lib
@jackl-xilinx @jgmelber should we start a top-level subfolder collecting all the kernels to avoid too much duplication?

Reply:
Hi @denolf - let me know if I should indeed bring `Ukernels` under it's own umbrella -> `aie_ukernel_lib` (I'm actually doing that while trying to install the binaries to `mlir-aie/install`).

This PR is a blocker for the initial demo of `i32` vanilla matmul (`matmul_scalar`) via iree-amd-aie - the `mm.cc` that I've added here is a bit different from other kernels - `"offset"` has been added to the APIs and accordingly the microkernel `matmul_scalar` has been modified.

Regarding all microkernels, I don't think it'd be a nice idea to plumb them down rightaway as part of this PR and can be handled perhaps in a subsequent set of PRs - the compilation and installation is being done only for the particular `mm.cc` (the one with "offsets").

Let me know your thoughts and I can address them soonest.

Reply:
Hi @Abhishek-Varma,

Let's indeed start by having the kernel for this PR in a "aie_kernels" subfolder, so that you can make progress.

As you point out, there are more aie hardcoded kernels on the mlir-aie repo that then can be moved in a subsequent PR.

Thanks for addressing this.